src/parser/jobs/dnc/modules/DirtyDancing.tsx

```diff
@@ -71,17 +75,28 @@ class Dance {
 		const actualFinish = _.last(this.rotation)
 		let expectedFinish = -1
 		if (actualFinish) {
-			if (TECHNICAL_FINISHES.includes(actualFinish.ability.guid)) {
+			if (this.initiatingStep.ability.guid === ACTIONS.TECHNICAL_STEP.id) {
 				expectedFinish =  ACTIONS.QUADRUPLE_TECHNICAL_FINISH.id
-			} else if (STANDARD_FINISHES.includes(actualFinish.ability.guid)) {
+			} else if (this.initiatingStep.ability.guid === ACTIONS.STANDARD_STEP.id) {
 				expectedFinish = ACTIONS.DOUBLE_STANDARD_FINISH.id
 			}
 		}
 		return expectedFinish
 	}
 
-	constructor(start: number) {
-		this.start = start
+	public get expectedEndTime(): number {
+		const actionData = getDataBy(ACTIONS, 'id', this.initiatingStep.ability.guid) as TODO
+		return this.start + actionData.gcdRecast * 1000
+			+ EXPECTED_DANCE_MOVE_COUNT[this.expectedFinishId] * DANCE_MOVE_COOLDOWN_MILLIS
+			+ 1000 // Additional 1s leniency to account for network latency
```

Comment:
Nit: this could (should?) be changed to a constant, maybe RUSHING_LATENCY_ALLOWANCE or something?

Reply:
Easy enough, done